jwst/extract_1d/soss_extract/soss_extract.py

```diff
@@ -321,6 +342,12 @@ def _build_tracemodel_order(engine, order_model, f_k, mask):
     mask : np.ndarray[bool]
         The global mask of pixels to be modeled. Bad pixels in the science data
         should remain unmasked.
+    gen_engine : bool, optional
+        Generate engine inside this function?  Alternative is to use or
+        generate the engine already associated with order_model.  This should
+        be True if reconstructing spectra within the Tikhonov tests, and False
+        if using the function within the main integation routine.
+        Default False.
```

Comment:
This doesn't seem to do precisely what is documented.  It appears that this flag forces re-computation of the engine even if one already exists, and also prevents the newly-computed engine from being saved into `order_model.orderengine`.  Is that the intended behavior?

If so, I'd suggest re-naming this something like `force_recompute_engine`, changing the logic a bit, and updating this docstring to specify that if True, this will _not_ modify `order_model.orderengine`